application/src/main/java/org/togetherjava/tjbot/commands/moderation/BanCommand.java

```diff
@@ -192,5 +200,22 @@ public void onSlashCommand(@NotNull SlashCommandEvent event) {
                 .requireNonNull(alreadyBanned.getFailure()), event, guild, target).orElseGet(
                         () -> banUserFlow(target, author, reason, deleteHistoryDays, guild, event));
         }).queue();
+
+        try {
+            database.write(context -> {
+                BanSystemRecord banSystemRecord = context.newRecord(BanSystem.BAN_SYSTEM)
+                    .setUserid(target.getIdLong())
+                    .setAuthorId(author.getIdLong())
+                    .setGuildId(guild.getIdLong())
+                    .setIsBanned(true)
+                    .setBanReason(reason);
+                if (banSystemRecord.update() == 0) {
+                    banSystemRecord.insert();
+                }
+            });
+            logger.info("Saved the user '{}' to the ban system.", target.getAsTag());
+        } catch (Exception exception) {
+            exception.printStackTrace();
```

Comment:
this is a heavy flaw.
1. never catch all exceptions - only the exceptions ur actually interested in
2. never just print stack trace in a proper app, ur bypassing the logging system completely
3. why even catching in the first place? just let it happen...